compiler/next/include/chpl/uast/Enum.h

```diff
@@ -68,37 +79,47 @@ class Enum final : public TypeDecl {
   ~Enum() override = default;
 
   static owned<Enum> build(Builder* builder, Location loc,
-                           UniqueString name, Decl::Visibility vis,
+                           owned<Attributes> attributes,
+                           Decl::Visibility vis,
+                           UniqueString name,
                            ASTList stmts);
 
   /**
     Return a way to iterate over the EnumElements and Comments.
    */
   ASTListIteratorPair<Expression> declOrComments() const {
-    return ASTListIteratorPair<Expression>(children_.begin(), children_.end());
+    auto begin = declOrCommentChildNum() >= 0
+          ? children_.begin() + declOrCommentChildNum()
+          : children_.end();
+    auto end = begin + numDeclOrComments();
+    return ASTListIteratorPair<Expression>(begin, end);
   }
 
   /**
    Return the number of EnumElements and Comments contained in this Enum.
    */
   int numDeclOrComments() const {
-    return this->numChildren();
+    return attributes() ? numChildren() - 1 : numChildren();
   }
   /**
    Return the i'th EnumElement or Comment in this Enum.
    */
   const Expression* declOrComment(int i) const {
-    const ASTNode* ast = this->child(i);
-    assert(ast->isExpression());
+    assert(i >= 0 && i < numDeclOrComments());
```

Comment:
FWIW with these I like `assert(0 <= i && i < numDeclOrComments())` better because it more closely matches the math `0 <= i < numDeclOrComments()` but I don't need to insist on following that in this or other PRs.